src/components/menu/menu.tsx

```diff
@@ -150,28 +179,181 @@ export class Menu {
                             'has-grid-layout': this.gridLayout,
                         }}
                     >
-                        <limel-menu-list
-                            class={{
-                                'has-grid-layout has-interactive-items':
-                                    this.gridLayout,
-                            }}
-                            items={this.items}
-                            type="menu"
-                            badgeIcons={this.badgeIcons}
-                            onSelect={this.handleSelect}
-                            ref={this.setListElement}
-                        />
+                        {this.renderBreadcrumb()}
+                        {this.renderMenuList()}
                     </limel-menu-surface>
                 </limel-portal>
             </div>
         );
     }
 
-    public componentDidRender() {
-        const slotElement = this.host.shadowRoot.querySelector('slot');
-        slotElement.assignedElements().forEach(this.setTriggerAttributes);
+    @Watch('items')
+    protected itemsWatcher() {
+        this.currentItems = null;
+
+        this.setFocus();
+    }
+
+    @Watch('currentItems')
+    protected currentItemsWatcher() {
+        this.setFocus();
+    }
+
+    @Watch('open')
+    protected openWatcher() {
+        if (!this.open) {
+            this.currentItems = null;
+
+            return;
+        }
+
+        this.setFocus();
     }
 
+    private renderBreadcrumb = () => {
+        if (!this.currentSubMenu) {
+            return;
+        }
+
+        const breadCrumbItems: MenuCrumbItem[] = [];
+        let currentItem = this.currentSubMenu;
+        while (currentItem) {
+            breadCrumbItems.unshift({
+                text: currentItem.text,
+                icon: {
+                    name: currentItem.icon,
+                    color: currentItem.iconColor,
+                },
+                menuItem: currentItem,
+            });
+            currentItem = currentItem.parentItem;
+        }
+
+        if (!breadCrumbItems.length) {
+            return;
+        }
+
+        return (
+            <limel-breadcrumbs
+                style={{
+                    'border-bottom': 'solid 1px rgb(var(--contrast-500))',
+                    'flex-shrink': '0',
+                }}
```

Comment:
Styles belong in the stylesheet 🙂

Reply:
Oh, or… these can't be in the stylesheet, because this stuff is lifted out to the portal, right?

That brings me to the point that this seems complex enough that it might warrant breaking it out into its own (internal) component. And it just so happens that doing so would let us avoid the inline styles as well!

Reply:
Yeah, me and Kia discussed that, but it would only be a very small wrapper of the limel-breadcrumbs which seems kind of risky for only 2 style properties.